From 7f0222445ac65d8d05e652cfe8c99f13a99e3244 Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Mon, 26 May 2025 12:52:32 +0300
Subject: [PATCH] [SW-229990] scripts: serdes: update rack config

SkyRiver3 includes 2 host nodes, each one connects to 4 nodes and each
node contains 4 OAMs. In total there are 32 OAMs.
They are all connected among each other by SerDes switches.
i.e. 16 OAMs per host -> 16 module ids. (4 bits)

Whitebox rack contains 16 nodes. On each such node there is a host
and 4 OAMs. In total there are 64 OAMs in a rack.
i.e. 4 OAMs per host -> 4 module ids. (2 bits)

This patch updates the serdes configuration accordingly.

Signed-off-by: Guy Nisan <gnisan@habana.ai>
Change-Id: Ia20cb88a73d135697d60aa4ae5740a03ee21c61f
---

diff --git a/scripts/serdes/gaudi3_serdes_config_flash_memory.json b/scripts/serdes/gaudi3_serdes_config_flash_memory.json
index 699e6b2..5aaab48 100644
--- a/scripts/serdes/gaudi3_serdes_config_flash_memory.json
+++ b/scripts/serdes/gaudi3_serdes_config_flash_memory.json
@@ -560,6 +560,104 @@
 					"enable_mask": "0x00FFFFFF",
 					"external_mask": "0x00FFFFFF",
 					"autoneg_mask": "0x0"
+				},
+				{
+					"index": 4,
+					"tx_pol": 0,
+					"rx_pol": 0,
+					"enable_mask": "0x00FFFFFF",
+					"external_mask": "0x00FFFFFF",
+					"autoneg_mask": "0x0"
+				},
+				{
+					"index": 5,
+					"tx_pol": 0,
+					"rx_pol": 0,
+					"enable_mask": "0x00FFFFFF",
+					"external_mask": "0x00FFFFFF",
+					"autoneg_mask": "0x0"
+				},
+				{
+					"index": 6,
+					"tx_pol": 0,
+					"rx_pol": 0,
+					"enable_mask": "0x00FFFFFF",
+					"external_mask": "0x00FFFFFF",
+					"autoneg_mask": "0x0"
+				},
+				{
+					"index": 7,
+					"tx_pol": 0,
+					"rx_pol": 0,
+					"enable_mask": "0x00FFFFFF",
+					"external_mask": "0x00FFFFFF",
+					"autoneg_mask": "0x0"
+				},
+
+				{
+					"index": 8,
+					"tx_pol": 0,
+					"rx_pol": 0,
+					"enable_mask": "0x00FFFFFF",
+					"external_mask": "0x00FFFFFF",
+					"autoneg_mask": "0x0"
+				},
+				{
+					"index": 9,
+					"tx_pol": 0,
+					"rx_pol": 0,
+					"enable_mask": "0x00FFFFFF",
+					"external_mask": "0x00FFFFFF",
+					"autoneg_mask": "0x0"
+				},
+				{
+					"index": 10,
+					"tx_pol": 0,
+					"rx_pol": 0,
+					"enable_mask": "0x00FFFFFF",
+					"external_mask": "0x00FFFFFF",
+					"autoneg_mask": "0x0"
+				},
+				{
+					"index": 11,
+					"tx_pol": 0,
+					"rx_pol": 0,
+					"enable_mask": "0x00FFFFFF",
+					"external_mask": "0x00FFFFFF",
+					"autoneg_mask": "0x0"
+				},
+
+				{
+					"index": 12,
+					"tx_pol": 0,
+					"rx_pol": 0,
+					"enable_mask": "0x00FFFFFF",
+					"external_mask": "0x00FFFFFF",
+					"autoneg_mask": "0x0"
+				},
+				{
+					"index": 13,
+					"tx_pol": 0,
+					"rx_pol": 0,
+					"enable_mask": "0x00FFFFFF",
+					"external_mask": "0x00FFFFFF",
+					"autoneg_mask": "0x0"
+				},
+				{
+					"index": 14,
+					"tx_pol": 0,
+					"rx_pol": 0,
+					"enable_mask": "0x00FFFFFF",
+					"external_mask": "0x00FFFFFF",
+					"autoneg_mask": "0x0"
+				},
+				{
+					"index": 15,
+					"tx_pol": 0,
+					"rx_pol": 0,
+					"enable_mask": "0x00FFFFFF",
+					"external_mask": "0x00FFFFFF",
+					"autoneg_mask": "0x0"
 				}
 			]
 		},
@@ -568,7 +666,7 @@
 			"type": 16,
 			"link_id": "0b01100",
 			"is_node_exists" : true,
-			"module_id_bits" : 4,
+			"module_id_bits" : 2,
 			"tx_lane_swap" : ["0x0", "0x0", "0x0", "0x0", "0x0", "0x0",
 			"0x0", "0x0", "0x0", "0x0", "0x0", "0x0"],
 			"configurations": [
